examples/otlp/foo_library/foo_library.cc

```diff
@@ -11,13 +11,9 @@ nostd::shared_ptr<trace::Tracer> get_tracer()
   return provider->GetTracer("foo_library");
 }
 
-// TODO: Remove all calls to span->End() once context memory issue is fixed
-// (https://github.com/open-telemetry/opentelemetry-cpp/issues/287)
-
 void f1()
 {
   auto span = get_tracer()->StartSpan("f1");
-  span->End();
```

Comment:
An explicit call to `End` is still necessary, not for the context memory issue, but because spans are not automatically ended anymore when the object goes out of scope (since `StartSpan` returns a `shared_ptr` instead of an `unique_ptr` now).

This should take care of ending the span:
```c++
 opentelemetry::trace::Scope scope(span);
```

Reply:
Just for spans in this file, even they are `shared_ptr`, there should be only one owner which is the declared `auto span`, so destruction of `auto span` will destruct the `Span` object and `End()` it accordingly. So current code should work.

Reply:
You're right. I still think we should add the `Scope`, otherwise spans will not be parented correctly (no current span will ever be set, so no parent will be set on any of the spans).

This was not broken by your change, that was already broken before.